spec/models/user_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+require "./app/models/user"
+
+describe User do
+  context "when all attributes are correct" do
+    it "is created" do
+      user = User.create(name: "John Doe",
+                         email: "user@gmail.com")
+
+      expect(user).to be_a User
+    end
+  end
+
+  context "when name not given" do
+    it "raises an error" do
+      user = User.create(email: "user@gmail.com")
+
+      expect { user.save! }
+        .to raise_error("Validation failed: Name must be entered")
+    end
+  end
+
+  context "when email not given" do
+    it "raises an error" do
+      user = User.create(name: "John Doe")
+
+      expect { user.save! }
+        .to raise_error("Validation failed: Email must be entered")
+    end
+  end
+
+  context "when name is not unique" do
+    it "raises an error" do
+      user = User.create(name: "John Doe",
+                         email: "user@gmail.com")
+      user.save!
```

Comment:
You are creating and saving. Two calls to the database in one. Line 36 is not needed.
But shoulda-matchers can simplify this a lot.